File: huzhe/2.1/insertSort.js
// 插入排序算法
function insertSort(arr) {
    for(let i = 1; i < arr.length; i++){
        let temp = arr[i];
        let j = i - 1;
        for(; j >= 0 && temp < arr[j]; j--){
            arr[j + 1] = arr[j];
        }
        arr[j + 1] = temp;
    }
    return arr;
}

console.log(insertSort([2,6,3,9,7,0,1,5,4]))

File: liyongpan/unit2/InsertSort.java
import java.util.Arrays;
import java.util.Collections;

public class InsertSort {

	public static void main(String[] args) {
		Integer[] a = randoms();
//		System.out.println(Arrays.toString(a));
		Arrays.sort(a, Collections.reverseOrder());
		System.out.println(Arrays.toString(a));
		long start = System.currentTimeMillis();
		System.out.println(start);
		for (int i = 0; i < a.length; i++) {
			for (int j = i + 1; j < a.length; j++) {
				if (a[j] < a[i]) {
					int temp = a[j];
					int k = j - i;
					while (k > 0) {
						a[k + i] = a[k + i - 1];
						k--;
					}
					a[i] = temp;
				}
			}
		}
		System.out.println(System.currentTimeMillis());
		System.out.println(System.currentTimeMillis() - start);
		System.out.println(Arrays.toString(a));
	}

	public static Integer[] randoms() {
		Integer[] arr = new Integer[10000];
		for(int i = 0, len = arr.length; i < len; i++) {
			int ran = (int) (Math.random() * 100000);
			arr[i] = ran;
		}
		return arr;
	}
}
